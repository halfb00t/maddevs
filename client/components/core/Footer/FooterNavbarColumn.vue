<template>
  <div
    class="footer-section"
    :class="`footer-section--${name}`"
  >
    <div class="footer-section__content">
      <div class="footer-section__menus">
        <FooterNavbarMenu
          v-for="(menu, idx) in menus"
          :key="menu.name || `menu-${idx}`"
          :menu-name="menu.name"
          :menu-routes="menu.routes"
        />
      </div>
    </div>
  </div>
</template>

<script>
import FooterNavbarMenu from '@/components/core/Footer/FooterNavbarMenu'

export default {
  name: 'FooterNavbarColumn',
  components: {
    FooterNavbarMenu,
  },

  props: {
    name: {
      type: String,
      default: '',
      required: true,
    },

    menus: {
      type: Array,
      default: () => ([]),
      required: true,
    },
  },

}
</script>
