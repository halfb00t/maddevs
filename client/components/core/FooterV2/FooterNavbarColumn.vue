<template>
  <div
    class="footer-section"
    :class="`footer-section--${name}`"
  >
    <div class="footer-section__content">
      <div class="footer-section__menus">
        <FooterNavbarMenu
          v-for="(menu, idx) in menus"
          :key="menu.name || `menu-${idx}`"
          :menu-name="menu.name"
          :menu-routes="menu.routes"
        />
      </div>
    </div>
  </div>
</template>

<script>
import FooterNavbarMenu from '@/components/core/FooterV2/FooterNavbarMenu'

export default {
  name: 'FooterNavbarColumn',
  components: {
    FooterNavbarMenu,
  },

  props: {
    isActive: {
      type: Boolean,
      default: false,
    },

    name: {
      type: String,
      default: '',
      required: true,
    },

    menus: {
      type: Array,
      default: () => ([]),
      required: true,
    },
  },

}
</script>

<style lang="scss" scoped>
</style>
