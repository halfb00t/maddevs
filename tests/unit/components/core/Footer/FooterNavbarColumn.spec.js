import { render } from '@testing-library/vue'
import { shallowMount } from '@vue/test-utils'

import FooterNavbarColumn from '@/components/core/Footer/FooterNavbarColumn'
import { footerNavigation } from './footerNavigationFixture'

jest.mock('~/helpers/generatorUid')

const stubs = ['FooterNavbarMenu']

const props = {
  menus: footerNavigation.company.menus,
  name: footerNavigation.company.name,
}
describe('FooterNavbarColumn', () => {
  it('should correct render footer navbar column', () => {
    const { container } = render(FooterNavbarColumn, {
      stubs,
      props,
    })
    expect(container)
      .toMatchSnapshot()
  })
  it('should correct render footer navbar column without data', () => {
    const wrapper = shallowMount(FooterNavbarColumn, {
      stubs,
      propsData: props,
    })
    expect(wrapper.vm.$options.props.menus.default.call())
      .toEqual([])
  })
})
