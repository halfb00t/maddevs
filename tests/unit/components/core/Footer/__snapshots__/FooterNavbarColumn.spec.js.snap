// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterNavbarColumn should correct render footer navbar column 1`] = `
<div>
  <div
    class="footer-section footer-section--company"
  >
    <div
      class="footer-section__content"
    >
      <div
        class="footer-section__menus"
      >
        <footernavbarmenu-stub
          menuname="Company"
          menuroutes="[object Object],[object Object],[object Object],[object Object],[object Object]"
        />
        <footernavbarmenu-stub
          menuname="Approach"
          menuroutes="[object Object],[object Object],[object Object],[object Object]"
        />
      </div>
    </div>
  </div>
</div>
`;
